test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/util/CollectionsOnListsTest.scala

```diff
@@ -7,97 +7,110 @@ import org.junit.Test
-    // Test: sort[T](List[T], Comparator[T])
-    def test[T](toElem: Int => T, cmpFun: (T, T) => Int,
-        absoluteOrder: Boolean = true): Unit = {
-      val list = factory.empty[T]
+  // Test: sort[T](List[T], Comparator[T])
+  def test_sort_with_comparator[T](factory: ListFactory, toElem: Int => T,
+      cmpFun: (T, T) => Int, absoluteOrder: Boolean = true): Unit = {
 
-      def testIfSorted(rangeValues: Boolean): Unit = {
-        for (i <- range.init)
-          assertTrue(cmpFun(list(i), list(i + 1)) <= 0)
-        if (absoluteOrder && rangeValues) {
-          for (i <- range)
-            assertEquals(0, cmpFun(list(i), toElem(i)))
-        }
-      }
+    val list = factory.empty[T]
 
-      val cmp = new ju.Comparator[T] {
-        override def compare(o1: T, o2: T): Int = cmpFun(o1, o2)
+    def testIfSorted(rangeValues: Boolean): Unit = {
+      for (i <- range.init)
+        assertTrue(cmpFun(list(i), list(i + 1)) <= 0)
+      if (absoluteOrder && rangeValues) {
+        for (i <- range)
+          assertEquals(0, cmpFun(list(i), toElem(i)))
       }
+    }
 
-      list.addAll(range.map(toElem))
-      ju.Collections.sort(list, cmp)
-      testIfSorted(true)
+    val cmp = new ju.Comparator[T] {
+      override def compare(o1: T, o2: T): Int = cmpFun(o1, o2)
+    }
+
+    list.addAll(range.map(toElem))
+    ju.Collections.sort(list, cmp)
+    testIfSorted(true)
+
+    list.clear()
+    list.addAll(range.reverse.map(toElem))
+    ju.Collections.sort(list, cmp)
+    testIfSorted(true)
 
+    for (seed <- List(0, 1, 42, -5432, 2341242)) {
+      val rnd = new scala.util.Random(seed)
       list.clear()
-      list.addAll(range.reverse.map(toElem))
+      list.addAll(range.map(_ => toElem(rnd.nextInt())))
       ju.Collections.sort(list, cmp)
-      testIfSorted(true)
-
-      for (seed <- List(0, 1, 42, -5432, 2341242)) {
-        val rnd = new scala.util.Random(seed)
-        list.clear()
-        list.addAll(range.map(_ => toElem(rnd.nextInt())))
-        ju.Collections.sort(list, cmp)
-        testIfSorted(false)
-      }
+      testIfSorted(false)
     }
+  }
+}
 
-    if (factory.sortableUsingCollections) { // Issue #2087
-      test[CustomComparable](new CustomComparable(_), (x, y) => x.compareTo(y), false)
-      test[Int](_.toInt, (x, y) => x.compareTo(y))
-      test[Long](_.toLong, (x, y) => x.compareTo(y))
-      test[Double](_.toDouble, (x, y) => x.compareTo(y))
+trait CollectionsOnListTest extends CollectionsOnCollectionsTest {
+
+  import CollectionsOnListTest._
+
+  def factory: ListFactory
+
+  @Test def sort_on_comparables(): Unit = {
+    if (factory.sortableUsingCollections) {
+      test_sort_on_comparables[CustomComparable](factory,
+          new CustomComparable(_), false)
+      test_sort_on_comparables[jl.Integer](factory, jl.Integer.valueOf)
+      test_sort_on_comparables[jl.Long](factory, _.toLong)
+      test_sort_on_comparables[jl.Double](factory, _.toDouble)
```

Comment:
Doesn't `CollectionsOnCopyOnWriteArrayListTestOnJDK8` perform exactly the same 4 calls (and same thing in `sort_with_comparator`)? Couldn't the 4 calls be abstracted away too? If not, this requires a `// !!!` comment on both sides to highlight the code duplication.